src/Parsers/Access/parseUserName.h

```diff
@@ -34,4 +34,9 @@ inline bool parseRoleNames(IParser::Pos & pos, Expected & expected, Strings & ro
     return parseUserNames(pos, expected, role_names);
 }
 
+inline bool parseStorageName(IParser::Pos & pos, Expected & expected, String & storage_name)
+{
+    return parseUserName(pos, expected, storage_name);
```

Comment:
A user name can contain `@`. It's not necessary to allow that for storage names.

Reply:
Yeah, I copied it from the `parseRoleName` above